tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckDates.java

```diff
@@ -680,6 +686,93 @@ && isTooMuchWiderThan(value, abbrValue)
+    private static final XPathParts dateParts =
+            XPathParts.getFrozenInstance(
+                    "//ldml/dates/calendars/calendar[@type=\"gregorian\"]/dateFormats/dateFormatLength[@type=\"short\"]/dateFormat[@type=\"standard\"]/pattern[@type=\"standard\"]");
+    private static final XPathParts availableParts =
+            XPathParts.getFrozenInstance(
+                    "//ldml/dates/calendars/calendar[@type=\"generic\"]/dateTimeFormats/availableFormats/dateFormatItem[@id=\"yyyyQQQ\"]");
+
+    private static final int calendarIndex = timeParts.findElement("calendar");
+    private static final int dateFormatLengthIndex = timeParts.findElement("dateFormatLength");
+    private static final int dateFormatIndex = timeParts.findElement("timeFormat");
+
+    // NOTE: the timeFormatLength is always at the same position as the dateFormatLength, but just
+    // for consistency...
+
+    private void checkDateTimeFormats(
+            String path, XPathParts parts, String value, List<CheckStatus> result) {
+
+        // Make sure that if the format has adjacent placeholders, we don't run numbers together
+        if (!value.contains("}{")) { // only worry about adjacents
+            return;
+        }
+        // we assume that relative dates don't end with digits
+        if ("relative".equals(parts.getAttributeValue(dateFormatIndex, "type"))) {
+            return;
+        }
+
+        String length = parts.getAttributeValue(dateFormatLengthIndex, "type");
+        String calendar = parts.getAttributeValue(calendarIndex, "type");
+
+        // By the spec, the dateTimeFormatLength will be the same as the dateFormatLength
+        // So see if the corresponding date ends with digits
+        // However, this only accounts for the stock formats; not combinations of available.
+
+        CLDRFile resolvedCldrFile = getResolvedCldrFileToCheck();
+
+        XPathParts currentDateParts =
+                dateParts
+                        .cloneAsThawed()
+                        .setAttribute(calendarIndex, "type", calendar)
+                        .setAttribute(dateFormatLengthIndex, "type", length);
+        String dateValue = resolvedCldrFile.getStringValue(currentDateParts.toString());
+        if (dateValue == null) {
+            throw new IllegalArgumentException(
+                    "Missing value: " + currentDateParts); // should never occur
+        }
+
+        // TODO: This is a first pass, to correct the bad cases.
+        // We should check later to see if we can allow more cases.
+        // IN THAT CASE, the following code could be the start of that, so it is left here, but
+        // inoperative
+
+        if (!DISABLE) {
+            // all current formats are date + time, not reverse
+            if (value.contains("{0}{1}")) {
+                throw new IllegalArgumentException("Enhance checks to handle {0}{1}");
+            }
+
+            // TODO: also check the available formats that are for a date
```

Comment:
This is an important TODO. We use these glue patterns to assemble date+time patterns from availableFormats.